kuiper-game/Domain/Ship.cs

```diff
@@ -1,50 +1,85 @@
 using System;
+using Kuiper.Domain.CelestialBodies;
 
 namespace Kuiper.Domain
 {
     public class Ship
     {
-        public Ship(string name, string shipClass, int speed)
+        private const double EARTH_GRAVITY = 9.80665;
+        public Ship(string name, ShipEngine engine, double dryMass)
         {
             Name = name;
-            ShipClass = shipClass;
-            Speed = speed;
+            Engine = engine;
+            DryMass = dryMass;
         }
 
         public string Name { get; }
+        public ShipEngine Engine { get; }
+        public CelestialBody CurrentLocation { get; set; }
+        public CelestialBody TargetLocation { get; set; }
+        public ShipStatus Status { get; set; }
+        public double WetMass 
+        { 
+            get 
+            {
+                return FuelMass + DryMass;
+            }
+        }
+        public double FuelMass { get; set;}
+        public double DryMass { get; }
 
-        public string ShipClass { get; }
-        public int Speed { get; }
-        public Location CurrentLocation { get; set; }
-        public Location TargetLocation { get; set; }
```

Comment:
Not quite sure how to make these not get;sets. In a perfect world they would only be changed by the SetCourseEvent and the ShipService. Any suggestions on how to do it?

Reply:
> Perhaps;
> Considering the ship an aggregate, which is sounds like is the case, would mean something along the lines of having a "ship.travel(destination)" api for instance. Here you would encapsulate the logic for how long it takes, taking mass and engine into account, perhaps even queuing a domain event. That could be acted upon somewhere else. This I could definitely see. Then it would indeed become the ships responsibility in my mind atleast, But it would inverse the current model - right?

Reply:
That's a good idea, so something like updating the shipservice (or a new navigation service?) to then know the ships current status, location and if it's moving somewhere.  That would pull that information out of the "car" object.